pkgs/tools/security/vault/default.nix

```diff
@@ -1,27 +1,35 @@
-{ stdenv, lib, fetchFromGitHub, buildGoPackage, installShellFiles, nixosTests
+{ stdenv, lib, fetchFromGitHub, buildGoModule, installShellFiles, nixosTests
 , makeWrapper
 , gawk
 , glibc
 }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "vault";
-  version = "1.7.3";
+  version = "1.8.0";
 
   src = fetchFromGitHub {
     owner = "hashicorp";
     repo = "vault";
     rev = "v${version}";
-    sha256 = "sha256-BO4xzZrX9eVETQWjBDBfP7TlD7sO+gLgbB330A11KAI=";
+    sha256 = "sha256-jyGymAplSe8vaSwbONtpRoAondxNBcRbFZ+YyOtr7qo=";
   };
 
-  goPackagePath = "github.com/hashicorp/vault";
+  vendorSha256 = "11mrdf7pwf8v7bb7wamv2yhixw4x06rd0qym942kk9bwif2balfp";
 
   subPackages = [ "." ];
 
   nativeBuildInputs = [ installShellFiles makeWrapper ];
 
-  buildFlagsArray = [ "-tags=vault" "-ldflags=-s -w -X ${goPackagePath}/sdk/version.GitCommit=${src.rev}" ];
+  preBuild = ''
+    buildFlagsArray+=(
+      "-tags=vault"
+      "-ldflags=\
+        -s -w \
+        -X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev} \
+        -X github.com/hashicorp/vault/sdk/version.Version=${version} \
+        -X github.com/hashicorp/vault/sdk/version.VersionPrerelease=")
```

Comment:
```suggestion
    buildFlagsArray+=(
      "-tags=vault"
    )

    ldflags = [
      "-s" "-w"
      "-X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev}"
      "-X github.com/hashicorp/vault/sdk/version.Version=${version}"
      "-X github.com/hashicorp/vault/sdk/version.VersionPrerelease="
```

Reply:
```
/nix/store/bmxafbqrlvcwpcwd21irmirid4rkpank-stdenv-linux/setup: line 91: ldflags: command not found
```

Reply:
Need to move them out of `preBuild`.
```diff
diff --git a/pkgs/tools/security/vault/default.nix b/pkgs/tools/security/vault/default.nix
index d502cc779c2..e5b8b370ff3 100644
--- a/pkgs/tools/security/vault/default.nix
+++ b/pkgs/tools/security/vault/default.nix
@@ -21,15 +21,14 @@ buildGoModule rec {
 
   nativeBuildInputs = [ installShellFiles makeWrapper ];
 
-  preBuild = ''
-    buildFlagsArray+=(
-      "-tags=vault"
-      "-ldflags=\
-        -s -w \
-        -X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev} \
-        -X github.com/hashicorp/vault/sdk/version.Version=${version} \
-        -X github.com/hashicorp/vault/sdk/version.VersionPrerelease=")
-  '';
+  buildFlagsArray = [ "-tags=vault" ];
+
+  ldflags = [
+    "-s" "-w"
+    "-X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev}"
+    "-X github.com/hashicorp/vault/sdk/version.Version=${version}"
+    "-X github.com/hashicorp/vault/sdk/version.VersionPrerelease="
+  ];
 
   postInstall = ''
     echo "complete -C $out/bin/vault vault" > vault.bash
```

Reply:
Done, but considering both uses are frequent enough in `nixpkgs`... :confused: